From 5cea20c678e3e2d4719fa830d8872630871aad53 Mon Sep 17 00:00:00 2001
From: Varun Kumar Reddy Yaparla <vyaparla@habana.ai>
Date: Tue, 04 Jan 2022 08:26:33 +0200
Subject: [PATCH] [SW-68660] gaudi2: use fake security defines from specs

This change uses fake security spi flash offset defines from
specs instead of local defines

Change-Id: I9211b0c2bac8c8d5e0900b7c887048ff00ab2c5c
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 4bdba2b..18bcb43 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -318,11 +318,6 @@
 					HL_SEC_ENGINE_PSOC);
 }
 
-#define SZ_1K			1024
-#define SZ_1M			(SZ_1K * SZ_1K)
-#define FAKE_OTP_OFFSET		(3 * SZ_1M)
-#define FAKE_FUSE_OFFSET	(3 * SZ_1M + 64 * SZ_1K)
-
 static const struct hl_sec_efuse *preboot_get_hl_sec_fake_efuse(void)
 {
 	struct hl_sec_efuse *fuse;
@@ -332,8 +327,8 @@
 	if (preboot_hl_sec_fuse_cache.is_invalid) {
 		fuse = &preboot_hl_sec_fuse_cache.fuse;
 		/* Copy efuse security data */
-		spi_flash_copy(&fuse->data, FAKE_FUSE_OFFSET + (EFUSE_SEC_OFFSET * sizeof(*val)),
-					sizeof(*fuse), "FAKE SECURITY FUSE");
+		spi_flash_copy(&fuse->data, FAKE_SECURITY_FUSE_OFFSET + (EFUSE_SEC_OFFSET *
+				sizeof(*val)), sizeof(*fuse), "FAKE SECURITY FUSE");
 
 		/* Fake fuse data is stored in negative logic */
 		for (i = 0; i < ARRAY_SIZE(fuse->data); i++) {
@@ -385,7 +380,7 @@
 
 static void preboot_security_copy_fake_otp(struct hl_sec_otp *otp)
 {
-	spi_flash_copy(otp, FAKE_OTP_OFFSET, sizeof(struct hl_sec_otp), "FAKE OTP");
+	spi_flash_copy(otp, FAKE_SECURITY_OTP_OFFSET, sizeof(struct hl_sec_otp), "FAKE OTP");
 }
 
 /* Number of DWORDS available in PSOC registers to store security key hash */
